06-classification_continued.Rmd

```diff
@@ -415,27 +507,32 @@ accuracy_vs_k <- ggplot(accuracies, aes(x = neighbors, y = mean)) +
   labs(x = "Neighbors", y = "Accuracy Estimate")
 accuracy_vs_k
 ```
-
-This visualization suggests that $K = 7$ provides the highest accuracy.
+Figure \@ref(fig:06-find-k) suggests that setting the number of 
+neighbours to $K =$ `r (accuracies %>% arrange(desc(mean)) %>% head(1))$neighbors`
+provides the highest accuracy.
 But as you can see, there is no exact or perfect answer here; 
- any selection between $K = 3$ and $13$ would be reasonably justified, as all
-of these differ in classifier accuracy by less than 1%. Remember: the
+ any selection from $K = 2$ and $15$ would be reasonably justified, as all
+of these differ in classifier accuracy by a small amount. Remember: the
 values you see on this plot are *estimates* of the true accuracy of our
-classifier. Although the $K=7$ value is higher than the others on this plot,
+classifier. Although the 
+$K =$ `r (accuracies %>% arrange(desc(mean)) %>% head(1))$neighbors` value is 
+higher than the others on this plot,
 that doesn't mean the classifier is actually more accurate with this parameter
 value! Generally, when selecting $K$ (and other parameters for other predictive
 models), we are looking for a value where:
 
 - we get roughly optimal accuracy, so that our model will likely be accurate
-- changing the value to a nearby one (e.g. from $K=7$ to 6 or 8) doesn't decrease accuracy too much, so that our choice is reliable in the presence of uncertainty
+- changing the value to a nearby one (e.g. adding or subtracting 1) doesn't decrease accuracy too much, so that our choice is reliable in the presence of uncertainty
 - the cost of training the model is not prohibitive (e.g., in our situation, if $K$ is too large, predicting becomes expensive!)
 
-### Under/overfitting
+### Under/Overfitting
 
-To build a bit more intuition, what happens if we keep increasing the number of neighbours $K$? In fact, the accuracy 
-actually starts to decrease! Rather than setting `grid = 10` and letting `tidymodels` decide what values of $K$ to try,
-let's specify the values explicitly by creating a data frame with a `neighbors` variable. 
-Take a look as the plot below as we vary $K$ from 1 to almost the number of observations in the data set:
+To build a bit more intuition, what happens if we keep increasing the number of
+neighbours $K$? In fact, the accuracy actually starts to decrease! Rather than
+setting `grid = 10` and letting `tidymodels` decide what values of $K$ to try,
+let's specify the values explicitly by creating a data frame with a `neighbors`
+variable. Figure \@ref(fig:06-lots-of-ks) shows a plot of estimated accuracy as 
+we vary $K$ from 1 to almost the number of observations in the data set.
 
 ```{r 06-lots-of-ks, fig.height = 4, fig.width = 5, fig.cap="Plot of accuracy estimate versus number of neighbours for many K values"}
 set.seed(1)
```

Comment:
In the next line where we do: `k_lots <- tibble(neighbors = seq(from = 1, to = 385, by = 10))` should we show the output of `k_lots`? I am not sure we have shown `seq` before in the book have we? Nor do we often use `tibble`?

Reply:
Also, the plot below makes the drop off in accuracy look super bad, it's not good, but it's not 0. We should make the y-axis go from 0-1.

Reply:
seq function is now explained earlier; i disagree with the plotting comment (makes it hard to see what's going on in the plot)